pkg/proxy/conntrackcleaner/cleaner.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conntrackcleaner
+
+import (
+	"k8s.io/kubernetes/pkg/proxy"
+	utilexec "k8s.io/utils/exec"
+)
+
+type Cleaner func(proxy.ServicePortMap, proxy.UpdateServiceMapResult, proxy.UpdateEndpointMapResult)
+
+// NewCleaner takes a mockable interface to os/exec and returns connection tracking entry cleaner.
+func NewCleaner(exec utilexec.Interface) Cleaner {
```

Comment:
Why not an interface?

Reply:
Yeah, returning an interface type would be more normal, even if it only has one function in it.
Although I'm not sure we even need an interface here; this could just be a handful of functions, which take a `utilexec.Interface` as an argument (like the existing `pkg/util/conntrack` functions).